### ISSUE TITLE
na_ontap_log_forward fails to create via REST

### ISSUE THREAD

**Issue by mamoep** [ansible-collections/netapp#82]
##### SUMMARY

na_ontap_log_forward fails to create the configuration via REST. 

##### ISSUE TYPE
- Bug Report

##### COMPONENT NAME

na_ontap_log_forward

##### ANSIBLE VERSION

```paste below
ansible 2.9.15
```

##### CONFIGURATION

```paste below

```

##### OS / ENVIRONMENT

ONTAP 9.7P8
RHEL 7.9 host
Ansible Collection netapp.ontap 21.2.0

##### STEPS TO REPRODUCE

Run a play with "na_ontap_log_forward".

```yaml
  - name: Set Log-Forwarding
    na_ontap_log_forward:
      state: present
      destination: "{{ log_forward.audit_log_host }}"
      port: "{{ log_forward.audit_log_port }}"
      protocol: "{{ log_forward.audit_log_protocol }}"
      hostname: "{{ inventory_hostname }}"
      <<: *login
    connection: local
```

##### EXPECTED RESULTS

Log-Forwarding should be created in ONTAP.

##### ACTUAL RESULTS

Tasks fails with "Cannot resolve the destination host...". There is no DNS issue on the cluster!

```paste below
TASK [Set Log-Forwarding] ********************************************************************************************************************************************************************************
task path: /ansible/private/playbooks/netapp/netapp_basic_setup.yml:153
<cluster.domain.local> ESTABLISH LOCAL CONNECTION FOR USER: username
<cluster.domain.local> EXEC /bin/sh -c 'echo ~username && sleep 0'
<cluster.domain.local> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo /home/username/.ansible/tmp `"&& mkdir "` echo /home/username/.ansible/tmp/ansible-tmp-1612513696.01-10699-58555074894331 `" && echo ansible-tmp-1612513696.01-10699-58555074894331="` echo /home/username/.ansible/tmp/ansible-tmp-1612513696.01-10699-58555074894331 `" ) && sleep 0'
Using module file /home/username/.ansible/collections/ansible_collections/netapp/ontap/plugins/modules/na_ontap_log_forward.py
<cluster.domain.local> PUT /home/username/.ansible/tmp/ansible-local-10579z14N5i/tmpppPE2s TO /home/username/.ansible/tmp/ansible-tmp-1612513696.01-10699-58555074894331/AnsiballZ_na_ontap_log_forward.py
<cluster.domain.local> EXEC /bin/sh -c 'chmod u+x /home/username/.ansible/tmp/ansible-tmp-1612513696.01-10699-58555074894331/ /home/username/.ansible/tmp/ansible-tmp-1612513696.01-10699-58555074894331/AnsiballZ_na_ontap_log_forward.py && sleep 0'
<cluster.domain.local> EXEC /bin/sh -c '/usr/bin/python /home/username/.ansible/tmp/ansible-tmp-1612513696.01-10699-58555074894331/AnsiballZ_na_ontap_log_forward.py && sleep 0'
<cluster.domain.local> EXEC /bin/sh -c 'rm -f -r /home/username/.ansible/tmp/ansible-tmp-1612513696.01-10699-58555074894331/ > /dev/null 2>&1 && sleep 0'
fatal: [cluster.domain.local]: FAILED! => {
    "changed": false,
    "invocation": {
        "module_args": {
            "cert_filepath": null,
            "destination": "loghost.domain.local",
            "facility": null,
            "feature_flags": {},
            "force": null,
            "hostname": "cluster.domain.local",
            "http_port": null,
            "https": true,
            "key_filepath": null,
            "ontapi": null,
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "port": 12345,
            "protocol": "tcp_encrypted",
            "state": "present",
            "use_rest": "auto",
            "username": "admin",
            "validate_certs": false,
            "verify_server": null
        }
    },
    "msg": {
        "code": "9240604",
        "message": "Cannot resolve the destination host  \"loghost.domain.local\". Check the DNS settings on this cluster."
    }
}
```

When using SSH CLI the command succeeds.  When using the parameter "use_rest: None" the task succeeds. 
```
changed: [cluster.domain.local] => {
    "changed": true,
    "invocation": {
        "module_args": {
            "cert_filepath": null,
            "destination": "loghost.domain.local",
            "facility": null,
            "feature_flags": {},
            "force": null,
            "hostname": "cluster.domain.local",
            "http_port": null,
            "https": true,
            "key_filepath": null,
            "ontapi": null,
            "password": "VALUE_SPECIFIED_IN_NO_LOG_PARAMETER",
            "port": 12345,
            "protocol": "tcp_encrypted",
            "state": "present",
            "use_rest": "None",
            "username": "admin",
            "validate_certs": false,
            "verify_server": null
        }
    }
}
```

**Comment by lonico** [comment]
Thank you for the detailed report and confirming it works with ZAPI and CLI.

The error message is coming directly from ONTAP, so it's possible REST is doing more validation than ZAPI or CLI.  Can you confirm that when the command successfully completes on ZAPI or CLI that loghost is actually reachable?    Logs should appear on loghost.

**Comment by mamoep** [comment]
I can confirm that the messages are forwarded to the loghost.

**Comment by lonico** [comment]
A suggestion is to use the `force` option.   I understand it does not solve the underlying issue with ONTAP.

**Comment by mamoep** [comment]
As mentioned in the bug it works with use_rest: none option. Can you check with the ONTAP people to get it fixed or do I have to open a support case?

**Comment by lonico** [comment]
Does it work for you with force?   There could still be an issue with the vserver DNS configuration.   But yes, it would better to  open a support case.

**Comment by mamoep** [comment]
NetApp Support opened BURT 1380288 for the issue.

**Comment by mamoep** [comment]
According to NetApp support portal, the issue is fixed in ONTAP 9.9.1RC1

**Comment by lonico** [comment]
ONTAP issue.
